tests/test_characters_counter.py

```diff
@@ -17,6 +20,34 @@ def test_count_characters_occurrence_raises_TypeError_when_unhashable_input_is_p
         with pytest.raises(TypeError):
             count_characters_occurrence(input_text)
 
+    @pytest.mark.parametrize(
+        "data_input, result, expectation",
+        [
+            ("Python is the best language in the world", {"t": 4, "e": 4, "h": 3}, does_not_raise()),
+            ("python", {"p": 1, "y": 1, "t": 1}, does_not_raise()),
+            ("tests\\text.txt", {"e": 80, "t": 72, "o": 64}, does_not_raise()),
+            ("tests\\text_2.txt", {}, pytest.raises(FileNotFoundError)),
+            (123, {}, pytest.raises(TypeError)),
+            ({}, {}, pytest.raises(TypeError)),
+            ("", {}, pytest.raises(ValueError)),
+        ],
+    )
+    def test_count_character_frequency(self, data_input: Any, result: Any, expectation: Type) -> None:
```

Comment:
Disagree with this test.
1. Separate positive and negative test cases.  Tests are typically not packaged into the production code so you can feel free to make a bit more tests and give them more descriptive name
2. Given When Then please